app/cdap/components/GridTable/index.tsx

```diff
@@ -276,16 +284,23 @@ export default function GridTable() {
     setRowsDataList(rowData);
   };
 
-  useEffect(() => {
-    getGridTableData();
-    setShowGridTable(Array.isArray(gridData?.headers) && gridData?.headers.length !== 0);
-  }, [gridData]);
-
   const handleColumnSelect = (columnName) => {
     setSelectedColumn((prevColumn) => (prevColumn === columnName ? '' : columnName));
     setColumnType(gridData?.types[columnName]);
   };
 
+  const onMenuOptionSelection = (option: string, supportedDataType: string[]) => {
+    setAddTransformationFunction({
+      option,
+      supportedDataType,
+    });
+  };
+
+  useEffect(() => {
+    getGridTableData();
+    setShowGridTable(Array.isArray(gridData?.headers) && gridData?.headers.length !== 0);
```

Comment:
I get that you just moved this function to be a bit later but theres no reason for showGridTable to be a state variable since it triggers a rerender here and is easily calculable.

Reply:
Not saying to change this since it isn't new code but its worth mentioning

Reply:
Sure Sean, point noted, we will take care of it